File: app/src/test/java/com/jameskbride/eventbusmvpdemo/security/SecurityErrorViewPresenterTest.java
package com.jameskbride.eventbusmvpdemo.security;

import org.greenrobot.eventbus.EventBus;
import org.junit.Before;
import org.junit.Test;
import org.mockito.Mock;

import static org.mockito.Mockito.verify;
import static org.mockito.MockitoAnnotations.initMocks;

public class SecurityErrorViewPresenterTest {

    @Mock private SecurityErrorViewPresenter.SecurityErrorView view;

    private EventBus eventBus;

    private SecurityErrorViewPresenter subject;

    @Before
    public void setUp() {
        initMocks(this);
        eventBus = EventBus.getDefault();

        subject = new SecurityErrorViewPresenter(eventBus);
        subject.view = view;
    }

    @Test
    public void onDismissItDismissesTheDialog() {
        subject.dismiss();

        verify(view).dismiss();
    }
}

File: app/src/test/java/com/jameskbride/eventbusmvpdemo/security/SecurityErrorViewPresenterTest.kt
package com.jameskbride.eventbusmvpdemo.security

import com.nhaarman.mockito_kotlin.verify
import org.greenrobot.eventbus.EventBus
import org.mockito.MockitoAnnotations.initMocks
import org.junit.Before
import org.junit.Test
import org.mockito.Mock

class SecurityErrorViewPresenterTest {

    @Mock private lateinit var view: SecurityErrorViewPresenter.SecurityErrorView

    private lateinit var eventBus: EventBus

    private lateinit var subject: SecurityErrorViewPresenter

    @Before
    fun setUp() {
        initMocks(this)
        eventBus = EventBus.getDefault()

        subject = SecurityErrorViewPresenter(eventBus)
        subject.view = view
    }

    @Test
    fun onDismissItDismissesTheDialog() {
        subject.dismiss()

        verify(view).dismiss()
    }
}